### DISCUSSION TITLE
LuaBridge3 performance

### DISCUSSION THREAD

**Discussion by Mellnik** [kunitoki/LuaBridge3#96] [General]
Speed is important for many applications. I ask myself where LuaBridge3 stands in comparison to vanilla LuaBridge and in comparison to other libraries.

For example [sol2 claims to be really fast](https://sol2.readthedocs.io/en/latest/benchmarks.html).
(https://github.com/ThePhD/lua-bindings-shootout)

It would be great to have benchmark tests. And if possible improvements in certain areas.

Let me know what you think!

**Comment by kunitoki** [comment]
Sounds good, even tho my focus now is to make it safe and secure to use. Later we can write some benchmarks.

**Comment by kunitoki** [comment]
I expect sol2 to beat LuaBridge (2 or 3 doesn't matter) in term of performance, and there's little we can do unless we rewrite some of the internals. But fwiw LuaBridge beats sol2 at compilation times, especially on huge bindings.

**Comment by Mellnik** [comment]
Agreed safety and security should be priority. Including good support on what is offered instead of having tons of poorly tested features.
 
Compile time is great. I have roughly 250 classes and 3000 members exposed with LB3 and it takes around 100 seconds to compile that part.

Part of my functions are from the Dear ImGui library where hundreds or thousands of static function calls happen each draw frame.

**Reply by rpatters1** [reply]
>I have roughly 250 classes and 3000 members exposed with LB3 and it takes around 100 seconds to compile that part.

My project is about the same size, maybe a bit larger. It builds in under 40 seconds on my MacStudio purchased last year. On my dinky Windows machine, however, 30 minutes! `;_;`

**Comment by kunitoki** [comment]
We are now in the same league as sol2 v3.0: 

<img width="630" height="470" alt="image" src="https://github.com/user-attachments/assets/cfb114d1-a83b-4511-949a-22ba4c2d34b6" />

https://github.com/kunitoki/LuaBridge3/pull/212

**Comment by kunitoki** [comment]
New numbers are scary

<img width="2100" height="4380" alt="image" src="https://github.com/user-attachments/assets/5c6d4bfe-1c59-481e-956f-86106fe27201" />
